modules/core/ante/ante.go

```diff
@@ -30,10 +32,19 @@ func (rrd RedundantRelayDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 		for _, m := range tx.GetMsgs() {
 			switch msg := m.(type) {
 			case *channeltypes.MsgRecvPacket:
-				response, err := rrd.k.RecvPacket(ctx, msg)
+				var (
+					response *channeltypes.MsgRecvPacketResponse
+					err      error
+				)
+				if ctx.IsReCheckTx() {
```

Comment:
I can add a comment about the ordering importance here, but the tests would catch if incorrectly switched to `IsCheckTx` first

Reply:
Up to you. Either way is fine by me!